### PR TITLE
WEB-341 Add Integration Tests for CreateClientComponent

### PR DESCRIPTION
**Pull request by gkbishnoi07**
## Description

Add Integration Tests for Create-Client.Component.ts file

## Related issues and discussion
[WEB-341](https://mifosforge.jira.com/browse/WEB-341?atlOrigin=eyJpIjoiMjAwOWY1NDJiOTVjNGYwYzg5Y2EyYWZhYmM2Mzg3MWQiLCJwIjoiaiJ9)

[WEB-341]: https://mifosforge.jira.com/browse/WEB-341?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

## Summary by CodeRabbit

- Tests
  - Added comprehensive integration tests for client creation covering validation, data aggregation, edge cases, and full submission flow.
  - Introduced a unified Jest test setup with environment and browser-API mocks, expanded test discovery to include .test files, and switched typings to Jest.

- Chores
  - Standardized Jest configuration with explicit options for coverage, transforms, module resolution, and test matching.
  - Aligned test scripts for local and CI workflows and upgraded test tooling (ts-node) for compatibility.

### CODE REVIEW

File: src/setup-jest.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+};
+
+Object.defineProperty(window, 'localStorage', {
+  value: localStorageMock
+});
+
+// Mock sessionStorage
+Object.defineProperty(window, 'sessionStorage', {
+  value: localStorageMock
+});
```
[the remark this answers is not in the document]
Comment (gkbishnoi07) [reply src/setup-jest.ts:29]: @coderabbitai is this a problem or just a suggestion?

**Comment by gkbishnoi07** [comment]
@IOhacker @alberto-art3ch @steinwinde please review my test file

**Review by steinwinde — APPROVED** [review]
LGTM
